tests/test-output-flex-way-change.cpp

```diff
@@ -0,0 +1,274 @@
+
+    options.append = true;
+
+    int num_t1 = 0;
+    int num_t2 = 0;
+    std::string update;
+    SECTION("to t1")
+    {
+        update.append("w10 v1 dV Tt1=yes Nn10,n11");
+        --num_t2;
+        ++num_t1;
+    }
+    SECTION("to t1 and t2")
+    {
+        update.append("w10 v1 dV Tt1=yes,t2=yes Nn10,n11");
+        ++num_t1;
+    }
+
+    REQUIRE_NOTHROW(db.run_import(options, update.c_str()));
+
+    CHECK(1 + num_t1 == conn.get_count("osm2pgsql_test_t1"));
+    CHECK(2 + num_t2 == conn.get_count("osm2pgsql_test_t2"));
+    CHECK(0 + num_t1 == conn.get_count("osm2pgsql_test_t1", "way_id = 10"));
+    CHECK(1 + num_t2 == conn.get_count("osm2pgsql_test_t2", "way_id = 10"));
+    CHECK(2 == conn.get_count("osm2pgsql_test_tboth"));
+}
+
+TEST_CASE("change way from t1 and t2")
+{
+    options_t options = testing::opt_t().slim().flex(conf_file);
+
+    testing::data_t data{tdata};
+    data.add({"w10 v1 dV Tt1=yes,t2=yes Nn10,n11",
+              "r30 v1 dV Tt=ag Mw10@mark,w11@,w12@mark,w13@,w14@mark"});
+
+    REQUIRE_NOTHROW(db.run_import(options, data()));
+
+    auto conn = db.db().connect();
+
+    CHECK(2 == conn.get_count("osm2pgsql_test_t1"));
+    CHECK(1 == conn.get_count("osm2pgsql_test_t1", "way_id = 10"));
+    CHECK(2 == conn.get_count("osm2pgsql_test_t2"));
+    CHECK(1 == conn.get_count("osm2pgsql_test_t2", "way_id = 10"));
+    CHECK(2 == conn.get_count("osm2pgsql_test_tboth"));
+
+    options.append = true;
+
+    int num_t1 = 0;
+    int num_t2 = 0;
+    std::string update;
+    SECTION("to t1")
+    {
+        update.append("w10 v1 dV Tt1=yes Nn10,n11");
+        --num_t2;
+    }
+    SECTION("to t2")
+    {
+        update.append("w10 v1 dV Tt2=yes Nn10,n11");
+        --num_t1;
+    }
+
+    REQUIRE_NOTHROW(db.run_import(options, update.c_str()));
+
+    CHECK(2 + num_t1 == conn.get_count("osm2pgsql_test_t1"));
+    CHECK(2 + num_t2 == conn.get_count("osm2pgsql_test_t2"));
+    CHECK(1 + num_t1 == conn.get_count("osm2pgsql_test_t1", "way_id = 10"));
+    CHECK(1 + num_t2 == conn.get_count("osm2pgsql_test_t2", "way_id = 10"));
+    CHECK(2 == conn.get_count("osm2pgsql_test_tboth"));
+}
+
+TEST_CASE("change tboth with stage2 to tboth without stage2")
+{
+    options_t options = testing::opt_t().slim().flex(conf_file);
+
+    testing::data_t data{tdata};
+    data.add({"w10 v1 dV Ttboth=yes Nn10,n11",
+              "r30 v1 dV Tt=ag Mw10@mark,w11@,w12@mark,w13@,w14@mark"});
+
+    REQUIRE_NOTHROW(db.run_import(options, data()));
+
+    auto conn = db.db().connect();
+
+    CHECK(1 == conn.get_count("osm2pgsql_test_t1"));
+    CHECK(1 == conn.get_count("osm2pgsql_test_t2"));
+    CHECK(3 == conn.get_count("osm2pgsql_test_tboth"));
+    CHECK(1 == conn.get_count("osm2pgsql_test_tboth",
+                              "way_id = 10 AND rel_ids = '{30}'"));
+
+    options.append = true;
+
+    REQUIRE_NOTHROW(db.run_import(
+        options, "r30 v2 dV Tt=ag Mw10@,w11@,w12@mark,w13@,w14@mark"));
```

Comment:
That boils to a relation change. Should already be covered by the relation change tests and can be removed here.

Reply:
okay, have remove it